broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -294,7 +291,15 @@ public HealthStatus getHealthStatus() {
 
   @Override
   public void addFailureListener(final FailureListener failureListener) {
-    actor.run(() -> this.failureListeners.add(failureListener));
+    actor.run(
+        () -> {
+          this.failureListeners.add(failureListener);
+          if (this.zeebePartitionHealth.getHealthStatus() == HealthStatus.HEALTHY) {
```

Comment:
Please use ZeebePartition.getHealthStatus() to determine the current health. 

`this.zeebePartitionHealth` only contains the health that the ZeebePartition can control (e.g. installation failure), while there are also other components that may fail, like the log stream and the stream processor. Each of these is a different component, and they are all registered in the `CriticalComponentsHealthMonitor`, which is used to determine the health status when you just call `ZeebePartition.getHealthStatus()`.

Reply:
Done